src/EditorFeatures/Next/NavigableSymbols/NavigableSymbolService.Source.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.GoToDefinition;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.CodeAnalysis.Editor.Host;
+
+namespace Microsoft.CodeAnalysis.Editor.NavigableSymbols
+{
+    internal partial class NavigableSymbolService : INavigableSymbolSourceProvider
+    {
+        private partial class Source : INavigableSymbolSource
+        {
+            private readonly ITextView _view;
+            private readonly ITextBuffer _buffer;
+            private readonly IEnumerable<Lazy<IStreamingFindUsagesPresenter>> _presenters;
+            private bool _disposed;
+            private readonly IWaitIndicator _waitIndicator;
+
+            public Source(
+                ITextView textView, 
+                ITextBuffer textBuffer, 
+                IEnumerable<Lazy<IStreamingFindUsagesPresenter>> streamingPresenters,
+                IWaitIndicator waitIndicator)
+            {
+                _view = textView;
+                _buffer = textBuffer;
+                _presenters = streamingPresenters;
+                _waitIndicator = waitIndicator;
+            }
+
+            public void Dispose()
+            {
+                _disposed = true;
+            }
+
+            public async Task<INavigableSymbol> GetNavigableSymbolAsync(SnapshotSpan triggerSpan, CancellationToken cancellationToken)
+            {
+                if (_disposed)
+                {
+                    return null;
+                }
+
+                var snapshot = triggerSpan.Snapshot;
+                var position = triggerSpan.Start;
+                var document = snapshot.GetOpenDocumentInCurrentContextWithChanges();
```

Comment:
based on a bug i just encountered with projection scenarios, can you check if document is null and bail if it is?